app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,47 @@
+class TasksController < ApplicationController
+  before_action :set_task, only: [:show, :destroy, :edit, :update]
+
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def edit
+  end
+
+  def update
+    if @task.update(task_params)
+      redirect_to task_path(@task), notice: 'Task has been updated'
+    end
```

Comment:
What happens if task is failed to update?

Reply:
Ooops, it should render edit page. I'll fix it